packages/aws-cdk-lib/aws-stepfunctions/lib/states/state.ts

```diff
@@ -438,6 +452,40 @@ export abstract class State extends Construct implements IChainable {
     });
   }
 
+  /**
+   * Render ItemProcessor in ASL JSON format
+   */
+  protected renderItemProcessor(): any {
+    if (!this.processor) return undefined;
```

Comment:
Same as above, do you think we should error here if `!this.processor && !this.iteration`?